go/vt/sidecardb/sidecardb.go

```diff
@@ -364,35 +396,44 @@ func (si *schemaInit) ensureSchema(table *sidecarTable) error {
 		ddlCount.Add(1)
 		return nil
 	}
-	log.Infof("Table schema was already up to date for the %s table in the %s sidecar database", table.name, SidecarDBName)
+	log.Infof("Table schema was already up to date for the %s table in the %s sidecar database", table.name, GetName())
 	return nil
 }
 
 // region unit-test-only
-// This section uses helpers used in tests, but also in the go/vt/vtexplain/vtexplain_vttablet.go.
+// This section uses helpers used in tests, but also in
+// go/vt/vtexplain/vtexplain_vttablet.go.
 // Hence, it is here and not in the _test.go file.
+const (
+	createTableRegexp = "(?i)CREATE TABLE .* `?\\_vt\\`?..*"
+	alterTableRegexp  = "(?i)ALTER TABLE `?\\_vt\\`?..*"
+)
 
-// Query patterns to handle in mocks.
-var sidecarDBInitQueries = []string{
-	ShowSidecarDatabasesQuery,
-	SelectCurrentDatabaseQuery,
-	CreateSidecarDatabaseQuery,
-	UseSidecarDatabaseQuery,
-}
-
-var sidecarDBInitQueryPatterns = []string{
-	CreateTableRegexp,
-	AlterTableRegexp,
-}
+var (
+	sidecarDBInitQueries = []string{
+		"use %s",
+		createSidecarDBQuery,
+		sidecarDBExistsQuery,
+	}
+	// Query patterns to handle in mocks.
+	sidecarDBInitQueryPatterns = []string{
+		createTableRegexp,
+		alterTableRegexp,
+	}
+)
 
-// AddSchemaInitQueries adds sidecar database schema related queries to a mock db.
+// AddSchemaInitQueries adds sidecar database schema related
+// queries to a mock db.
+// This is for unit tests only!
 func AddSchemaInitQueries(db *fakesqldb.DB, populateTables bool) {
+	once.Do(loadSchemaDefinitions)
 	result := &sqltypes.Result{}
+	db.AddQuery(fmt.Sprintf("use %s", GetIdentifier()), result)
 	for _, q := range sidecarDBInitQueryPatterns {
 		db.AddQueryPattern(q, result)
 	}
 	for _, q := range sidecarDBInitQueries {
-		db.AddQuery(q, result)
+		db.AddQuery(fmt.Sprintf(q, GetIdentifier()), result)
```

Comment:
Continued - again I feel like this is not idiomatic; we should use `ParseAndBind` instead of `fmt.Sprintf`